src/nvim/os/stdpaths.c

```diff
@@ -69,13 +69,19 @@ const char *get_appname(void)
   return env_val;
 }
 
-/// Ensure that APPNAME is valid. In particular, it cannot contain directory separators.
+// Ensure that APPNAME is valid. In particular, it cannot contain .. surrounded
+// by path separators, the start of the string, or the end of the string.
 bool appname_is_valid(void)
 {
   const char *appname = get_appname();
   const size_t appname_len = strlen(appname);
+  if (appname[0] == PATHSEP) {
+    return false;
+  }
   for (size_t i = 0; i < appname_len; i++) {
```

Comment:
is the for-loop still needed?

Reply:
The for loop is specifically for `..` as although absolute paths can't be used you can do the equivalent by setting appname to `../../../../../effectively/absolute/path`

Reply:
I double checked just in case, the new test will fail if the for loop is taken out